API-Samples/separate_image_sampler/separate_image_sampler.cpp

```diff
@@ -304,7 +304,11 @@ int sample_main(int argc, char *argv[]) {
     vkDestroySampler(info.device, separateSampler, NULL);
     vkDestroyImageView(info.device, info.textures[0].view, NULL);
     vkDestroyImage(info.device, info.textures[0].image, NULL);
-    vkFreeMemory(info.device, info.textures[0].mem, NULL);
+    vkFreeMemory(info.device, info.textures[0].image_memory, NULL);
+    if (info.textures[0].needs_staging) {
```

Comment:
if these are safely initialized/reset to NULL don't need to make this conditional.